geth/rpc/client_test.go

```diff
@@ -44,6 +45,51 @@ func (s *RPCTestSuite) SetupTest() {
 	s.NodeManager = nodeManager
 }
 
+func (s *RPCTestSuite) TestNewClient() {
+	require := s.Require()
+
+	config, err := MakeTestNodeConfig(params.RinkebyNetworkID)
+	require.NoError(err)
+
+	nodeStarted, err := s.NodeManager.StartNode(config)
+	require.NoError(err)
+	require.NotNil(config)
+
+	<-nodeStarted
+
+	node, err := s.NodeManager.Node()
+	require.NoError(err)
+
+	// upstream disabled, local node ok
+	_, err = rpc.NewClient(node, config.UpstreamConfig)
+	require.NoError(err)
+
+	// upstream enabled with incorrect URL, local node ok
+	upstreamBad := config.UpstreamConfig
+	upstreamBad.Enabled = true
+	upstreamBad.URL = "///__httphh://///incorrect_urlxxx"
+	_, err = rpc.NewClient(node, upstreamBad)
+	require.NotNil(err)
+	require.IsType(rpc.UpstreamError{}, err)
+
+	// upstream enabled with correct URL, local node ok
+	upstreamGood := config.UpstreamConfig
+	upstreamGood.Enabled = true
+	upstreamGood.URL = "http://example.com/rpc"
+	_, err = rpc.NewClient(node, upstreamGood)
+	require.Nil(err)
+
+	// upstream disabled, local node failed (stopped)
+	nodeStopped, err := s.NodeManager.StopNode()
+	require.NoError(err)
+
+	<-nodeStopped
+
+	_, err = rpc.NewClient(node, config.UpstreamConfig)
+	require.NotNil(err)
+	require.IsType(rpc.LocalError{}, err)
+}
```

Comment:
This test contains a lot of different tests. Please, wrap them with `t.Run` or separate them.

Reply:
`t.Run`, probably not a case when using `testify`, but let me see what can be done.

Reply:
Yeah. Sorry, it's too late for good comments.

Reply:
Please, forget it for now.